src/main/scala/com/gu/fastly/Lambda.scala

```diff
@@ -175,14 +175,15 @@ class Lambda {
   }
 
   private def extractAliasPaths(event: Event): Seq[String] = {
-    event.payload.flatMap { payload =>
-      payload match {
-        case EventPayload.DeletedContent(deleted) => deleted.aliasPaths
-        case EventPayload.Content(content) => content.aliasPaths
-        case EventPayload.RetrievableContent(retrievable) => retrievable.aliasPaths
-        case _ => None
-      }
-    }.getOrElse(Seq.empty)
+    def getPaths(maybeAliases: Option[Seq[AliasPath]]): Seq[String] = {
+      maybeAliases.fold(Seq.empty[String]) { _.map(_.path) }
+    }
+    event.payload.fold(Seq.empty[String]) {
```

Comment:
getOrElse is often easier to read than fold in a situation like this as it lets you write the happy path first. 
Personal perference so not blocking.